src/Stratis.Bitcoin.Features.Miner/BlockDefinition.cs

```diff
@@ -471,7 +471,7 @@ private void OnlyUnconfirmed(TxMempool.SetEntries testSet)
         /// <summary>
         /// Test if a new package would "fit" in the block.
         /// </summary>
-        private bool TestPackage(long packageSize, long packageSigOpsCost)
+        protected virtual bool TestPackage(TxMempoolEntry entry, long packageSize, long packageSigOpsCost)
```

Comment:
There are also classes inheriting `BlockDefinition` in the smart contracts feature. We will possibly want the future drift logic in the PoS version of them?

Reply:
Smart contracts are POA no future drift ASAIK right @noescape00 ?

Reply:
Note that the bugfix rule does appear in the smart contracts PoS rule registrations, but the PoA version might have a different intended design.

Reply:
Our PoS implementation is currently broken in other areas. So no worries - it would be nice if you could chuck in our SmartContractPosPowBlockDefinition a `TODO: If reimplementing Smart contract PoS look at time drift` but otherwise we won't be using that for now so doesn't affect us.

PoA I'm not sure about - @noescape00 if you could let us know if PoA requires anything here I'm happy to update